packages/flutter/lib/src/rendering/shifted_box.dart

```diff
@@ -617,19 +653,33 @@ class RenderConstrainedOverflowBox extends RenderAligningShiftedBox {
     );
   }
 
+  // If passthrough, the size will be as small as its child when non-overflowing,
+  // thus is cannot be sizedByParent.
   @override
-  bool get sizedByParent => true;
+  bool get sizedByParent => fit != OverflowBoxFit.passthrough;
```

Comment:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values

Reply:
Also elsewhere in this PR where you compare with the enum.